extensions/amp-story/1.0/test/test-amp-story.js

```diff
@@ -164,6 +164,32 @@ describes.realWin('amp-story', {
         });
   });
 
+  it('should pause the story when bookend becomes active', () => {
```

Comment:
Sorry to be picky about this, but I feel like these tests belongs in the `test-store-service` file.

To keep tests independent and a great coverage without too much complexity, I think what matters is that:
- We test that `TOGGLE_BOOKEND` also updates `PAUSED_STATE` ([example](https://github.com/ampproject/amphtml/blob/master/extensions/amp-story/1.0/test/test-amp-story-store-service.js#L167))
- Then, `PAUSED_STATE` and each component's behavior is tested separately. This logic you're testing is actually already well covered [here](https://github.com/ampproject/amphtml/blob/master/extensions/amp-story/1.0/test/test-amp-story.js#L533-L615)

This way, store's logic stays in the store's tests. And in `test-amp-story`, we simply test how the story component reacts to individual store state updates.

Reply:
That's true... we should probably talk about our integration tests in general. e.g. When to write integration tests? Where should they live? Etc...